spec/models/piece_spec.rb

```diff
@@ -1,11 +1,41 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  before do
+  before :each do
     # Note: We want the board to be emptry for this test.
     allow_any_instance_of(Game).to receive(:populate_board!).and_return true
   end
 
+  describe '#move_to!' do
+    before :each do
+      @piece_white = FactoryGirl.create(:piece, row: 0, column: 0, color: 'white')
+      @piece_black = FactoryGirl.create(:piece, row: 0, column: 7, color: 'black')
+    end
+    it 'successfully moved the pieces to destination' do
+      piece = FactoryGirl.create(:piece, row: 0, column: 3)
+      # byebug
+
+      expect(piece.move_to!(0, 7)).to eq(true)
+    end
+
+    it 'successfully captured a piece' do
+      expect(@piece_white.move_to!(0, 7)).to eq(true)
+    end
+    
+    it 'should return false because piece is our own' do
+      
+      white_piece = FactoryGirl.create(:piece, row: 0, column: 7, color: 'white')
+      
+      expect(@piece_white.move_to!(0, 7)).to eq(false)
```

Comment:
You have the variables mixed up here. I think you want to be using `white_piece`